server/src/test/java/org/elasticsearch/common/blobstore/fs/FsBlobContainerTests.java

```diff
@@ -165,6 +167,35 @@ public void testCompareAndExchange() throws Exception {
         );
     }
 
+    public void testAtomicWriteMetadataWithoutAtomicOverwrite() throws IOException {
+        this.fileSystem = new FilterFileSystemProvider("nooverwritefs://", fileSystem) {
+            @Override
+            public void move(Path source, Path target, CopyOption... options) throws IOException {
+                final Set<CopyOption> optionSet = Set.of(options);
+                if (optionSet.contains(StandardCopyOption.REPLACE_EXISTING)
+                    && optionSet.contains(StandardCopyOption.ATOMIC_MOVE)
+                    && Files.exists(target)) {
+                    // simulate a file system that can't do atomic move + overwrite
+                    throw new IOException("no atomic overwrite moves");
+                } else {
+                    super.move(source, target, options);
+                }
+            }
+        }.getFileSystem(null);
+        final String blobName = randomAlphaOfLengthBetween(1, 20).toLowerCase(Locale.ROOT);
+        final byte[] blobData = randomByteArrayOfLength(randomIntBetween(1, 512));
+        PathUtilsForTesting.installMock(fileSystem); // restored by restoreFileSystem in ESTestCase
+        final Path path = PathUtils.get(createTempDir().toString());
+        Files.write(path.resolve(blobName), blobData);
```

Comment:
Can we use `container.writeBlobAtomic` here too with `failIfAlreadyExists = true`? And maybe verify that it doesn't leave the temporary blob in place?